flink-state-backends/flink-statebackend-changelog/src/test/java/org/apache/flink/state/changelog/ChangelogStateBackendTestUtils.java

```diff
@@ -158,16 +155,15 @@ public static void testMaterializedRestore(
             keyedBackend.setCurrentKey(2);
             state.update(new StateBackendTestBase.TestPojo("u2", 2));
 
-            awaitMaterialization(keyedBackend, env.getMainMailboxExecutor());
+            // trigger materialization manually
+            periodicMaterializationManager.triggerMaterialization();
 
             keyedBackend.setCurrentKey(2);
             state.update(new StateBackendTestBase.TestPojo("u2", 22));
 
             keyedBackend.setCurrentKey(3);
             state.update(new StateBackendTestBase.TestPojo("u3", 3));
 
-            awaitMaterialization(keyedBackend, env.getMainMailboxExecutor());
-
```

Comment:
Could you explain this change? Shouldn't we make sure that materialization is completed at this point?

Reply:
Now everything happens in a single main thread, including the async phase.

No need of awaiting any more

Reply:
I don't see any changes related to  threads in this PR, could you elaborate?

Reply:
Why is the test single thread? Because the materialization is triggered manually from the main thread now, the mailbox executor is sycn mailbox and async pool is a direct executor. Now everything is executed in a single thread. This is easier to reason about for a unit test. I am writing a ITTest as well. So multi-thread will be tested in a separate ticket.

Reply:
I think it's very fragile to rely on 
```
this.asyncOperationsThreadPool = Executors.newDirectExecutorService();
```
in constructor of a different class (`MockEnvironment`) in a different module.
That class is also used in many other tests. This change adds an **implicit** contract that its default `asyncOperationsThreadPool` **must** be direct.

Could you explain what was the motivation for the change in the first place?